torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -704,8 +693,17 @@ PjRtComputationClient::ExecuteReplicated(
   // Required as of cl/518733871
   execute_options.use_major_to_minor_data_layout_for_callbacks = true;
 
-  std::optional<std::vector<xla::PjRtFuture<xla::Status>>> returned_futures(
-      devices.size());
+  // Grab the shared lock and block the `WaitDeviceOps` until buffer is
+  // ready. Since this is the SPMD code path. There is no points to grab
+  // devices lock for every individual device.
+  TF_VLOG(5) << "ExecuteReplicated acquiring PJRT device lock for "
+             << spmd_device_str;
+  auto op = operation_tracker_.StartOperation(spmd_device_str);
```

Comment:
nit, can we rename this to `op_tracker`? `op` is a really confusing name, I thought it is the operations that needs to be executed when I read the `OnReady` code down below.

Reply:
Now that we've solved the easy problems (synchronizing concurrent operations between devices), let's move on to the hard one: naming things.

I don't think `op_tracker` works here, because the parent owner of all of the operation counters is called `OperationTracker`. It seems conventional in C++ to name RAII resources directly after what they represent. E.g., all of the lock objects are just calling `lock` on a mutex, but they don't hold the lock state themselves. I do get the confusion though.

Reply:
oh I see.. it is actually an op.... it is very confusing through haha

Reply:
I'm okay calling this resource (currently `Operation`) an `OperationTracker`. But, then we need a new name for `OperationTracker`. Perhaps `OperationTrackerTracker`?

Reply:
lol `OperationTrackerManager`?

Reply:
Oof, "`TrackerManager`" gives me Java vibes

Reply:
Settled for "OperationManager" and "OperationTracker" as the names. Let me know if that works for you.